### PR TITLE
A clarification on cross entropy loss function calculations

### PR DESCRIPTION
**Pull request by arun-r80**
Hi Mr Nielsen, 

I am raising this PR, as an attempt to clarify a question I had on  your valuable book. 
This is specifically about cross entropy loss function, wherein you had detailed that the error (dC/dw(i,j) is equal to a(i) - y(i) for last layer of the network. 
While I cannot agree more to the conclusion, I see that your code uses training output y for the "expected" output y(i) while calculating the error for output layer. I see that we need to propose a mechanism to arrive at expected value of output for last layer, and I see your proposition/suggestion/design that the neural layer will fire only in the neuron with index corresponding to the training output, and all other neurons will have expected value as zero. 

My code change corresponds to this assumption, and I fee the original code might not provide needed result. 
I will upgrade the code to latest Python version and explore this further. 

I would really appreciate if you could clarify this for me, 

Regards
Arun R

### CODE REVIEW

**Comment by arun-r80** [comment]
I understand the structure of y now. I am closing this PR
